mantle/kola/tests/ignition/kdump.go

```diff
@@ -0,0 +1,222 @@
+                if hfp.Service == "ssh" {
+                    address = "10.0.2.2"
+                    port = fmt.Sprintf("%d",hfp.HostPort)
+                }
+            }
+        default:
+            m, err = pc.NewMachine(ignition)
+            address = m.IP()
+            port = "22"
+        }
+        if err != nil {
+            c.Fatal(err)
+        }
+
+    // insert indentation in front of the entries
+    // to avoid errors in the butane file
+    var padded = ""
+    for _, line := range strings.Split(strings.TrimSuffix(string(PrivKeyBuf), "\n"), "\n") {
+        padded = fmt.Sprintf("%s          %s\n", padded, line)
+    }
+
+    return SshServer {
+        Machine:    m,
+        MachineAddress:    address,
+        SSHPort: port,
+        PubSSH: string(pubkeyBuf),
+        PrivSSH: padded,
+    }
+}
+
+func kdumpSSHTest(c cluster.TestCluster) {
+    ssh_host := setupSSHMachine(c)
+    butane := conf.Butane(fmt.Sprintf(`variant: fcos
+version: 1.5.0
+storage:
+  files:
+    - path: /root/.ssh/id_ssh_kdump.pub
+      mode: 0600
+      contents:
+        inline: |
+          %s
+    - path: /root/.ssh/id_ssh_kdump
+      mode: 0600
+      contents:
+        inline: |
+%s
+    - path: /root/.ssh/config
+      mode: 0644
+      overwrite: true
+      contents:
+        inline: |
+           Host %s
+               StrictHostKeyChecking no
+               Port %s
+    - path: /etc/kdump.conf
+      overwrite: true
+      contents:
+        inline: |
+          ssh core@%s
+          sshkey /root/.ssh/id_ssh_kdump
+          path /home/core/crash
+          core_collector makedumpfile -F -l --message-level 1 -d 31
+systemd:
+  units:
+    - name: kdump.service
+      enabled: true
+kernel_arguments:
+    should_exist:
+      - crashkernel=512M`,
+       ssh_host.PubSSH, ssh_host.PrivSSH, ssh_host.MachineAddress, ssh_host.SSHPort, ssh_host.MachineAddress))
+
+    opts := platform.MachineOptions{
+        MinMemory: 2048,
+    }
+
+    kdump_machine, err := c.NewMachineWithOptions(butane, opts)
+    if err != nil {
+        c.Fatalf("Unable to create test machine: %v", err)
+    }
+
+    // Wait for kdump to become active
+    // 3 minutes should be enough to generate the kdump initramfs
+    err = util.Retry(12, 15*time.Second, func() error {
+
+        kdump_status, err:= c.SSH(kdump_machine, "systemctl is-active kdump.service")
+
+        if err != nil {
+           return err
+        } else if string(kdump_status) == "inactive" {
+           err = errors.New(fmt.Sprintf("Kdump.service is not ready: %s.", string(kdump_status)))
+           return err
+        }
+        return nil
+    })
+    if err != nil {
+        c.Fatal("Timed out while waiting for kdump.service to be ready.")
+    }
+
+    // crash the kernel
+    // We ignore the error as this commands always hangs (it crashes the kernel !)
+    _, _= c.SSH(kdump_machine, "sudo su -c \"echo 'c' > /proc/sysrq-trigger\"")
```

Comment:
How about:

```suggestion
    _, err = c.SSH(kdump_machine, "sudo systemd-run sh -c 'sleep 5 && echo c > /proc/sysrq-trigger'")
    if err != nil {
        return fmt.Errorf("failed to queue kernel crash: %w", err)
    }
```

Reply:
Ah cool ! I was thinking of making a systemd unit to trigger the crash to alleviate this, but opted not to as it would have been weird. Your solution is nicer. I wasn't aware of systemd-run. Thanks !